src/EditorFeatures/CSharpTest/MisplacedUsingDirectives/MisplacedUsingDirectivesInCompilationUnitCodeFixProviderTests.cs

```diff
@@ -0,0 +1,488 @@
+        {
+            var testCode = $@"using System;
+
+{typeDefinition}
+";
+
+            return VerifyAnalyzerAsync(testCode, InsideNamespaceOption, DiagnosticResult.EmptyDiagnosticResults);
+        }
+
+        /// <summary>
+        /// Verifies that having using statements in the compilation unit will not produce any diagnostics when there are attributes present.
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnitWithAttributes_ValidUsingStatements()
+        {
+            var testCode = @"using System.Reflection;
+
+[assembly: AssemblyVersion(""1.0.0.0"")]
+
+namespace TestNamespace
+{
+    using System;
+    using System.Threading;
+}
+";
+
+            return VerifyAnalyzerAsync(testCode, InsideNamespaceOption, DiagnosticResult.EmptyDiagnosticResults);
+        }
+
+        /// <summary>
+        /// Verifies that the code fix will move the using directives and not place System directives first.
+        /// </summary>
+        [Fact]
+        public Task WhenInsidePreferred_UsingsInCompilationUnit_UsingsMovedAndSystemPlacedFirstIgnored()
+        {
+            var testCode = @"using Microsoft.CodeAnalysis;
+using SystemAction = System.Action;
+using static System.Math;
+using System;
+
+using static System.String;
+using MyFunc = System.Func<int, bool>;
+
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Foo
+{
+    public class Bar
+    {
+    }
+}
+";
+
+            var fixedTestCode = @"namespace Foo
+{
+    using Microsoft.CodeAnalysis;
+    using SystemAction = System.Action;
+    using static System.Math;
+    using System;
+
+    using static System.String;
+    using MyFunc = System.Func<int, bool>;
+
+    using System.Collections.Generic;
+    using System.Collections;
+
+    public class Bar
+    {
+    }
+}
+";
+
+            var expected = new DiagnosticResult[]
+            {
+                Diagnostic(MisplacedUsingDirectivesInCompilationDiagnosticAnalyzer.InsideDescriptor).WithLocation(1, 1),
+                Diagnostic(MisplacedUsingDirectivesInCompilationDiagnosticAnalyzer.InsideDescriptor).WithLocation(2, 1),
+                Diagnostic(MisplacedUsingDirectivesInCompilationDiagnosticAnalyzer.InsideDescriptor).WithLocation(3, 1),
+                Diagnostic(MisplacedUsingDirectivesInCompilationDiagnosticAnalyzer.InsideDescriptor).WithLocation(4, 1),
+                Diagnostic(MisplacedUsingDirectivesInCompilationDiagnosticAnalyzer.InsideDescriptor).WithLocation(6, 1),
+                Diagnostic(MisplacedUsingDirectivesInCompilationDiagnosticAnalyzer.InsideDescriptor).WithLocation(7, 1),
+                Diagnostic(MisplacedUsingDirectivesInCompilationDiagnosticAnalyzer.InsideDescriptor).WithLocation(9, 1),
+                Diagnostic(MisplacedUsingDirectivesInCompilationDiagnosticAnalyzer.InsideDescriptor).WithLocation(10, 1),
+            };
```

Comment:
these tests are basically not readable due to having to go manually map from this to the actual code to determine what they mean.  The test is no longer declarative with the inputs/outputs being enough to gauge if things are appropraite.  for example i can't see if any of those spans are wrong trivially by examining the test.

Reply:
Updated.